UserAgent/Search.swift

```diff
@@ -7,68 +7,105 @@
 //
 
 import Foundation
+import Shared
 
-class Search {
+public class Search {
 
-}
+    public enum AdultFilterMode: String, CaseIterable {
+        case liberal
+        case conservative
+    }
 
-extension Search: BrowserCoreClient {
-    private static let defaultConfig = Config(selected: "de", available: ["de"])
+    public struct Country {
+        public var key: String
+        public var name: String
+    }
 
     public struct Config {
-        public var selected: String
-        public var available: [String]
+        public var selected: Country
+        public var available: [Country]
     }
 
-    public enum AdultFilterMode: String, CaseIterable {
-        case liberal
-        case conservative
-    }
+    private static let defaultRegion = Country(key: "de", name: Strings.SettingsSearchResultForGerman)
+
+    static let defaultConfig = Config(selected: Search.defaultRegion, available: [Search.defaultRegion])
+
+}
+
+extension Search: BrowserCoreClient {
 
     public static func getBackendCountries(callback: @escaping (Config) -> Void) {
-        browserCore.callAction(
+        self.browserCore.callAction(
             module: "search",
             action: "getBackendCountries",
             args: []
         ) { (error, result) in
             if error != nil {
-                callback(Self.defaultConfig)
+                DispatchQueue.main.async {
+                    callback(Self.defaultConfig)
+                }
                 return
             }
 
             guard let backends = result as? [String: [String: Any]] else {
-                callback(Self.defaultConfig)
+                DispatchQueue.main.async {
```

Comment:
Would be better for the callback provider to deal the the queue.

Reply:
Done